tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreManagedLedgerOffloader.java

```diff
@@ -218,12 +228,19 @@ public String getOffloadDriverName() {
                         }
                         entryBytesWritten += blockStream.getBlockEntryBytesCount();
                         partId++;
+                        this.mxBean.recordOffloadBytes(extraMetadata.get(MANAGED_LEDGER_NAME),
+                                blockStream.getBlockEntryBytesCount());
                     }
 
                     dataObjectLength += blockSize;
                 }
 
+                long startUploadTime = System.nanoTime();
                 writeBlobStore.completeMultipartUpload(mpu, parts);
+                this.mxBean.recordWriteToStorageLatency(extraMetadata.get(MANAGED_LEDGER_NAME),
+                        System.nanoTime() - startUploadTime, TimeUnit.NANOSECONDS);
+                this.mxBean.recordOffloadTime(extraMetadata.get(MANAGED_LEDGER_NAME),
+                        System.nanoTime() - start, TimeUnit.NANOSECONDS);
```

Comment:
Maybe we should record once offload time after writing the index.

Reply:
seems you are right